### PR TITLE
Suppress buttons when job approved

### PR DESCRIPTION
**Pull request by nathanaelhoun**
## Changes

- [x] Suppress the button "Approve job" after it has been clicked
- [x] Updated README.md (gif updated)

## Changelog

N.A.

### CODE REVIEW

File: server/plugin/approve.go [hunk 1]
```diff
@@ -25,9 +25,27 @@ func (p *Plugin) httpHandleApprove(w http.ResponseWriter, r *http.Request) {
...
+			filteredAttach := attach
+			attach.Actions = nil
+			for _, action := range attach.Actions {
+				if action.Id != "approvecirclecijob" {
+					filteredAttach.Actions = append(filteredAttach.Actions, action)
+				}
+			}
+
+			filteredAttach.Color = "#50F100" // green
+			filteredAttach.Title = "This CircleCI workflow have been approved"
```
Comment (pradeepmurugesan) [inline server/plugin/approve.go]: Is it possible to add who approved?

Comment (nathanaelhoun) [reply server/plugin/approve.go]: image
Of course! Just pushed :-)

File: server/plugin/approve.go [hunk 2]
```diff
@@ -22,12 +22,37 @@ func (p *Plugin) httpHandleApprove(w http.ResponseWriter, r *http.Request) {
...
 	if appErr != nil {
 		p.API.LogError("Unable to get post", "postID", requestData.PostId)
-	} else if _, appErr := p.API.UpdatePost(originalPost); appErr != nil {
-		// TODO : remove the button
-		p.API.LogError("Unable to update post", "postID", originalPost.Id)
+	} else {
+		newAttachments := []*model.SlackAttachment{}
+		for _, attach := range originalPost.Attachments() {
+			filteredAttach := attach
+			attach.Actions = nil
```
Comment (nathanaelhoun) [inline server/plugin/approve.go]:
```suggestion
			filteredAttach.Actions = nil
```

File: server/plugin/approve.go [hunk 3]
```diff
@@ -22,12 +22,37 @@ func (p *Plugin) httpHandleApprove(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	username := userID
```
Comment (pradeepmurugesan) [inline server/plugin/approve.go]: We can have this as empty and show it as approved by only if we were able to resolve the user id to a user name.

Approved by xchq1244 is not a good user experience.

Personally I feel we should leave it as approved if we couldn't get the user name. Not ablocker
 Approved

**Review by pradeepmurugesan — APPROVED** [review]

File: server/plugin/approve.go (hunk 3 above)
Comment (nathanaelhoun) [reply server/plugin/approve.go]: You are right. Changed it :-)

**Pull request merged**